scripts/plotting/cloud_regime_analysis.py

```diff
@@ -0,0 +1,1369 @@
+                        # Deleting the variables we want to keep in our dataset, all remaining variables will be dropped upon opening the files, this allows for faster opening of large files
+                        remove.remove(var)
+                        # If there's a LANDFRAC variable keep it in the dataset
+                        landfrac_present = True
+                        try: remove.remove('LANDFRAC')
+                        except: landfrac_present = False
+                        
+                        # Opening dataset and dropping irrelevant data
+                        ds_b = xr.open_mfdataset(files, drop_variables = remove)
+
+                    # If variables are not present in h0 tell the user the variables do not exist, and that there is not COSP output for this data
+                    except: 
+                        print(f'  {var} does not exist in h0 files, does this run have {data} COSP output? Skipping {data} for now')
+                        missing_var = True # used to skip the code below and move onto the next var name
+
+                # Executing further analysis on this data 
+                finally:
+
+                    # Skipping var if its not present in data files
+                    if missing_var:
+                        continue
+
+                    # Adjusting lon to run from -180 to 180 if it doesnt already
+                    if np.max(ds_b.lon) > 180: 
+                        ds_b.coords['lon'] = (ds_b.coords['lon'] + 180) % 360 - 180
+                        ds_b = ds_b.sortby(ds_b.lon)
+                
+                    # Selecting only points over ocean or points over land if only_ocean_or_land has been used
+                    if only_ocean_or_land != False:
+                        # If LANDFRAC variable is present, use it to mask
+                        if landfrac_present == True:
+                            if only_ocean_or_land == 'L': ds_b = ds_b.where(ds_b.LANDFRAC == 1)
+                            elif only_ocean_or_land == 'O': ds_b = ds_b.where(ds_b.LANDFRAC == 0)
+                        # Otherwise use cartopy
+                        else:
+                            land = create_land_mask(ds_b)
+                            dims = ds_b.dims
+
+                            # Inserting new axis to make land a broadcastable shape with ds_b
+                            for n in range(len(dims)):
+                                if dims[n] != 'lat' and dims[n] != 'lon':
+                                    land = np.expand_dims(land, n)
+
+                            # Masking out the land or water
+                            if only_ocean_or_land == 'L': ds_b = ds_b.where(land == 1)
+                            elif only_ocean_or_land == 'O': ds_b = ds_b.where(land == 0)
+                            else: raise Exception('Invalid option for only_ocean_or_land: Please enter "O" for ocean only, "L" for land only, or set to False for both land and water')
```

Comment:
Don't want to kill the ADF here:

```suggestion
                            else:
                                print('Invalid option for only_ocean_or_land: Please enter "O" for ocean only, "L" for land only, or set to False for both land and water')
                                return
```